### PR TITLE
HugsLib like options API

### LINKED ISSUE #61: Suggestion: Mod Settings Framework
Hi, I would like to suggest a mod settings framework that other mods can use.

As an example I would like to recommend Hugslib from the Rimworld Community, as I think it's quite simple and effective to use:

usage example:
https://github.com/Fumblesneeze/RimBetterSpeed/blob/master/Source/BetterSpeed/Mod.cs

source is here:
https://github.com/UnlimitedHugs/RimworldHugsLib/tree/master/Source/Settings

the idea is that you as a user implement a base class where you give it a displayname and then you can add SettingHandle<T> fields to that class and initialize them with whatever selection of controls seems good, e.g. slider, textfield, number, and the framework collects all of these on one mod settings page, generates the UI and calls the SettingsChanged() callbacks when settings have been loaded or changed by the user. Reading the settings is easy since the SettingsHandle<T> will implicitly convert to to T.

### PR DESCRIPTION
**Pull request by Tyler-IN**
Begins effort for https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/issues/61

### CODE REVIEW

File: src/CommunityPatch/CommunityPatchSubModule.cs
Comment (tynakuh) [inline src/CommunityPatch/CommunityPatchSubModule.cs:30]: I'd prefer explicitly enforcing a file type/extension here.   Given they're all toml files as well.

File: src/CommunityPatch/Options/OptionsStore.cs
Comment (tynakuh) [inline src/CommunityPatch/Options/OptionsStore.cs:9]: Can we name these variables to be clearer for API purposes?

**Review by tynakuh — COMMENTED** [review]
LGTM to me otherwise once we have a second pair of eyes.

File: src/CommunityPatch/Options/OptionsStore.cs
Comment (Tyler-IN) [reply src/CommunityPatch/Options/OptionsStore.cs:9]: `ns` to `namespace`?

File: src/CommunityPatch/CommunityPatchSubModule.cs
Comment (Tyler-IN) [reply src/CommunityPatch/CommunityPatchSubModule.cs:30]: The other simple key-value-pair bannerlord config files are all .txt files, apart from that there's xml files.

File: src/CommunityPatch/Options/OptionsStore.cs
Comment (tynakuh) [reply src/CommunityPatch/Options/OptionsStore.cs:9]: Yep

File: src/CommunityPatch/CommunityPatchSubModule.cs
Comment (tynakuh) [reply src/CommunityPatch/CommunityPatchSubModule.cs:30]: Ah I see the intention now, resolving.

File: src/CommunityPatch/Options/OptionsStore.cs
Comment (Tyler-IN) [reply src/CommunityPatch/Options/OptionsStore.cs:13]: `namespace` is a language keyword.

Comment (Tyler-IN) [reply src/CommunityPatch/Options/OptionsStore.cs:13]: Discussed in Discord, will add xml docs to flesh out description of parameters.

**Comment by Tyler-IN** [comment]
> Un-needed. Actually exists in Apply already.

I think at some point we'll be displaying some GUI for patches, applicable patches will need to not show as skipped when applicable unless something went wrong when applying.
In this case, after being applied at campaign start up, it'd show as not applicable and applied, but skipped.

**Comment by Tyler-IN** [comment]
This has been forked off to be made into a separate module.

https://github.com/Tyler-IN/MnB2-Bannerlord-ModOptions

CommunityPatch will add it as a dependecy and use it.

**Comment by Tyler-IN** [comment]
This has references to options that need to be refactored and merged once ModOptions is up and running.

File: src/CommunityPatch/ContractAnnotationAttribute.cs
Comment (tryadelion) [inline src/CommunityPatch/ContractAnnotationAttribute.cs:22]: remove uneccessary spaces

File: src/CommunityPatch/Options/CustomOptionCategoryViewModel.cs
Comment (tryadelion) [inline src/CommunityPatch/Options/CustomOptionCategoryViewModel.cs:26]: remove unnecessary spaces

**Review by tryadelion — APPROVED** [review]
Just some unnecessary trailing whitespaces / empty ines.

**Pull request closed without merging**
